Wrap generate_binstubs in an if statement

Padgett opened the pull request:
Add a default node attribute ['brightbox-ruby']['generate_binstubs'] set to true

There are situations when node['brightbox-ruby']['gems'] will be empty or not include rubygems-bundler, in those cases attempting to generate binstubs for rubygems-bundler will throw an exception.

Commit:
Wrap generate_binstubs in an if statement
Add a default node attribute ['brightbox-ruby']['generate_binstubs'] set to true

There are situations when node['brightbox-ruby']['gems'] will be empty, in those cases attempting to generate binstubs for rubygems-bundler will throw an exception.

benlangfeld commented:
This only gets run when rubygems-bundler is installed. Can you please show a high verbosity Chef log showing the failure?

Padgett commented:
We are actually failing during spec:

```
1) cia_infra::ruby converges successfully
     Failure/Error: expect { chef_run }.not_to raise_error

       expected no Exception, got #<Chef::Exceptions::ResourceNotFound: Cannot find a resource matching gem_package[rubygems-bundler] (did you define it first?)> with backtrace:
         # /tmp/d20160511-24078-9so616/cookbooks/brightbox-ruby/recipes/default.rb:51:in `block in from_file'
         # /tmp/d20160511-24078-9so616/cookbooks/brightbox-ruby/recipes/default.rb:49:in `from_file'
         # /tmp/d20160511-24078-9so616/cookbooks/cia_infra/recipes/ruby.rb:5:in `from_file'
         # ./spec/unit/recipes/ruby_spec.rb:12:in `block (2 levels) in <top (required)>'
         # ./spec/unit/recipes/ruby_spec.rb:16:in `block (3 levels) in <top (required)>'
         # ./spec/unit/recipes/ruby_spec.rb:16:in `block (2 levels) in <top (required)>'
         # ./spec/unit/recipes/ruby_spec.rb:16:in `block (2 levels) in <top (required)>'
Finished in 28.83 seconds (files took 2.16 seconds to load)
42 examples, 1 failure

Failed examples:

rspec ./spec/unit/recipes/ruby_spec.rb:15 # cia_infra::ruby converges successfully
```

Padgett commented:
It looks like the failure is ocurring https://github.com/mojolingo/brightbox-ruby-cookbook/blob/develop/recipes/default.rb#L51 where it's looking for the `gem_package[rubygems-bundler]` resource

benlangfeld commented:
I take it you're not including rubygems-bundler in the list of installed gems? Maybe make this conditional on the presence of that gem in the list?

Padgett commented:
You are correct.
I will do that and push another commit.

Commit:
Remove generate_binstubs node attribute
Replace rubygems-bundler generate binstubs attribute check with a look at the gems array

benlangfeld commented:
Thanks!

Padgett commented:
Thank You! 💯

cwebberOps commented:
@benlangfeld would it be possible for you to cut a release to the supermarket?

benlangfeld commented:
Published as 1.2.2, @cwebberOps @Padgett
